ivy/container/gradients.py

```diff
@@ -1,9 +1,310 @@
+        map_sequences: bool = False,
+    ) -> ivy.Container:
+        return ContainerBase.multi_map_in_static_method(
+            "lars_update",
+            ws,
+            dcdws,
+            lr,
+            decay_lambda=decay_lambda,
+            inplace=inplace,
+            stop_gradients=stop_gradients,
+            key_chains=key_chains,
+            to_apply=to_apply,
+            prune_unapplied=prune_unapplied,
+            map_sequences=map_sequences,
+        )
+
+    def lars_update(
+        self,
+        dcdws,
+        lr,
+        decay_lambda=0,
+        inplace=None,
+        stop_gradients=True,
+        key_chains: Optional[Union[List[str], Dict[str, str]]] = None,
+        to_apply: bool = True,
+        prune_unapplied: bool = False,
+        map_sequences: bool = False,
+    ):
+        return self.static_lars_update(
+            self,
+            dcdws,
+            lr,
+            decay_lambda,
+            inplace,
+            stop_gradients,
+            key_chains,
+            to_apply,
+            prune_unapplied,
+            map_sequences,
+        )
+
+    @staticmethod
+    def static_adam_update(
+        ws,
+        dcdws,
+        lr,
+        mw_tm1,
+        vw_tm1,
+        step,
+        beta1=0.9,
+        beta2=0.999,
+        epsilon=1e-7,
+        inplace=None,
+        stop_gradients=True,
+        key_chains: Optional[Union[List[str], Dict[str, str]]] = None,
+        to_apply: bool = True,
+        prune_unapplied: bool = False,
+        map_sequences: bool = False,
+    ) -> ivy.Container:
+        return ContainerBase.multi_map_in_static_method(
+            "adam_update",
+            ws,
+            dcdws,
+            lr,
+            mw_tm1=mw_tm1,
+            vw_tm1=vw_tm1,
+            step=step,
+            beta1=beta1,
+            beta2=beta2,
+            epsilon=epsilon,
+            inplace=inplace,
+            stop_gradients=stop_gradients,
+            key_chains=key_chains,
+            to_apply=to_apply,
+            prune_unapplied=prune_unapplied,
+            map_sequences=map_sequences,
+        )
+
+    def adam_update(
+        self,
+        dcdws,
+        lr,
+        mw_tm1,
+        vw_tm1,
+        step,
+        beta1=0.9,
+        beta2=0.999,
+        epsilon=1e-7,
+        inplace=None,
+        stop_gradients=True,
+        key_chains: Optional[Union[List[str], Dict[str, str]]] = None,
+        to_apply: bool = True,
+        prune_unapplied: bool = False,
+        map_sequences: bool = False,
+    ):
+        return self.static_adam_update(
+            self,
+            dcdws,
+            lr,
+            mw_tm1,
+            vw_tm1,
+            step,
+            beta1,
+            beta2,
+            epsilon,
+            inplace,
+            stop_gradients,
+            key_chains,
+            to_apply,
+            prune_unapplied,
+            map_sequences,
+        )
+
+    @staticmethod
+    def static_lamb_update(
+        ws,
```

Comment:
these should not be plural either, `ws` should be `w` etc. These must match the names of the functional API in order for the nestable variant of the functions to work correctly. The arguments and keyword arguments are copied directly to this static method, and so the argument names must match exactly. The instance methods should also use the same non-plural argument names.